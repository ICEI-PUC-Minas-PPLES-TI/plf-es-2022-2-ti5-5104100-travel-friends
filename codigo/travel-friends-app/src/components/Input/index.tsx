import React from "react";
import { View, Text, TextInput } from "react-native";

import { styles } from "./styles";

type InputProps = {
  keyboardType?: "default" | "decimal-pad" | "email-address";
  text: string;
  value: string;
  onChangeText: (text: any) => void;
  onBlur?: () => void;
};

export default function Input({
  text,
  value,
  onChangeText,
  onBlur,
  keyboardType,
}: InputProps) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{text}</Text>
      <TextInput
        keyboardType={keyboardType}
        value={value}
        style={styles.input}
        onChangeText={onChangeText}
        onBlur={onBlur}
      />
    </View>
  );
}
