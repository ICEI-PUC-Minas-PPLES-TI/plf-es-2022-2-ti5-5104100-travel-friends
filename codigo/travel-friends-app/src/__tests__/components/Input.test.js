import React from "react";
import renderer from "react-test-renderer";

import Input from "../../components/Input";

describe("<Input />", () => {
  it("has 1 child", () => {
    const tree = renderer
      .create(<Input text={""} onChangeText={""} value={""} />)
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(<Input text={""} onChangeText={""} value={""} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
