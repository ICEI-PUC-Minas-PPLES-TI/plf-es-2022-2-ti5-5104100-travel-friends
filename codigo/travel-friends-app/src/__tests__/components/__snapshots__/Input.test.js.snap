// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#818191",
      }
    }
  />
  <TextInput
    onChangeText=""
    secureTextEntry={false}
    style={
      Object {
        "borderColor": "#818191",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginTop": 4,
        "padding": 6,
      }
    }
    value=""
  />
</View>
`;
